### PR TITLE
oc-calendar: Make invitation/update mails have the event name as subject

### PR DESCRIPTION
**Pull request by icandigitbaby**
SOGo event notification mails add information to the event name in their
subject (e.g. _Event invitation: "foo"_). The client uses the mail
subject to create a copy of the event on the attendee's calendar, so we
need to strip out that extra information if we want the event to have
the proper name.

Suggested `NEWS` line:
* Accepted & updated event names are now shown correclty

### CODE REVIEW

File: OpenChange/MAPIStoreMailMessage.m [hunk 1]
```diff
@@ -538,16 +541,45 @@ - (int) getPidTagNormalizedSubject: (void **) data
...
 
   subject = [self subject];
-  colIdx = [subject rangeOfString: @":"].location;
-  if (colIdx != NSNotFound && colIdx < 4)
-    stringValue = [[subject substringFromIndex: colIdx + 1]
-                    stringByTrimmingLeadSpaces];
+  if (mailIsMeetingRequest)
+    {
+      NSUInteger quoteStartIdx, quoteEndIdx;
+      NSRange quoteRange;
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreMailMessage.m]: The declaration of variables within a if clause (frame) is not part of the coding style of this library. Although I like this style, uniform throughout the code wins in this case.

File: OpenChange/MAPIStoreMailMessage.m [hunk 2]
```diff
@@ -537,17 +540,44 @@ - (int) getPidTagSubjectPrefix: (void **) data
...
+
+      /* As specified in [MS-OXCMAIL] 2.2.3.2.6.1, if there are three
+         or less characters followed by a colon at the beginning of
+         the subject, we can assume that's the subject prefix */
+      colIdx = [subject rangeOfString: @":"].location;
+      if (colIdx != NSNotFound && colIdx < 4)
+        {
+          stringValue = [[subject substringFromIndex: colIdx + 1]
+                            stringByTrimmingLeadSpaces];
+        }
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreMailMessage.m]: Curly braces here are not required. One-line statement :nail_care:

**Comment by blaxter** [comment]
:+1:

**Comment by sixstone-qq** [comment]
:+1:

**Pull request merged**